pkg/resource/interfaces.go

```diff
@@ -96,3 +120,35 @@ type Managed interface {
 
 	Bindable
 }
+
+// A Policy is a Kubernetes object representing... TODO(hasheddan)
+type Policy interface {
+	runtime.Object
+	metav1.Object
+
+	DefaultClassReferencer
+}
+
+// A PolicyList is a Kubernetes object representing... TODO(hasheddan)
+type PolicyList interface {
+	runtime.Object
+	metav1.ListInterface
+
+	PolicyLister
+}
+
+// A ClusterPolicy is a Kubernetes object representing... TODO(hasheddan)
+type ClusterPolicy interface {
```

Comment:
It is not lost on me that both `Policy` and `ClusterPolicy` are the same interface (also true for `PolicyList` and `ClusterPolicyList`). The reason for separate implementation is that they do exist as separate API kinds so having separate interfaces is useful for clear testing, especially if differences between policies and cluster policies arise in the future.